GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1165,6 +1166,10 @@ private void UpdatePlaceholders()
                             {
                                 this.AddParentFoldersToListToKeep(folderPlaceholder.Path, folderPlaceholdersToKeep);
                             }
+                            else
+                            {
+                                ++deleteFolderPlaceHolder;

```

Comment:
I would move this down after line 1182 since that is where we are actually deleting the placeholder from the database.  Also if we want as much information as possible I would log the `deleteFolderPlaceholderAttempted` and have a counter for the ReExpanded folders so we know how many of those as well?